### PR TITLE
Investigate deploy pipelines state after move

### LINKED ISSUE #25: Add continuous deployment into separate production cluster
**Goal**

Also continuously deploy production at `auctions.makerdao.network`.

**Context**

sidestream-tech/auction-ui#363 set the baseline helm charts for continuous deployments and added the continuous deployment to `main.auction-ui.k8s.sidestream.tech`.

Merging sidestream-tech/auction-ui#363 will show if that actually worked, the goal if this issue is to then add a step to continuously deploy into the production cluster on a tag event (triggered via a github release for example).

**Tasks**

- [x] resolve any problems that occured during the deployment triggered by the merge of sidestream-tech/auction-ui#363
- [ ] add a continuous deployment step into the production cluster

### LINKED ISSUE #56: Re-enable CD pipelines after repo switch
Depends on #48

**Goal**

Also re-enable CD pipelines, now that CI pipelines are secured and work

**Context**

In #48 we made sure that forks cannot (easily) leak credentials via drone pipelines. Now that this was ensured, we can re-enable the continuous deployment pipelines into our cluster.

**Tasks**

- [x] re-enable `main.auction-ui.k8s` pipeline
- [x] #25

### PR DESCRIPTION
**Pull request by BracketJohn**
Closes #56
Closes #25 

This PR adds a continuous deployment of https://production.auction-ui.k8s.sidestream.tech. It is deemed to take over the role of https://auctions.makerdao.network. Right now the deployment is in "stealth mode" - it doesn't consume any API quota or tweet any tweets.

See #25 for the environment limitations. 

In order to test this, the deployment of this env was enabled to be "on push" instead of "on tag". This should be reverted before the merge.

### CODE REVIEW

**Comment by BracketJohn** [comment]
@valiafetisov see:
- the deployment process here: https://drone.sidestream.tech/sidestream-tech/unified-auctions-ui/91/1/3,
- to check the logs of the deployment:
  - [visit this page](http://localhost:8001/api/v1/namespaces/kubernetes-dashboard/services/https:kubernetes-dashboard:/proxy/#/pod?namespace=auction-ui-production)
  - follow [these instructions to log in](https://github.com/sidestream-tech/infrastructure/blob/0817c05b21b5cf47ecfd0b68ae7deff06e80279a/README.md#kubernetes-dashboard)
    -  it should work with the credentials you already have in your aws-vault (: 

@LukSteib/ @florianstoecker please checkout the deployment here: https://production.auction-ui.k8s.sidestream.tech

**Comment by valiafetisov** [comment]
Checking `bot-twitter` logs I can only see errors like:
```
ERROR [InternalError: Internal error] {
code: [33m-32603[39m
}
```
http://localhost:8001/api/v1/namespaces/kubernetes-dashboard/services/https:kubernetes-dashboard:/proxy/#/log/auction-ui-production/bot-twitter-8577c9b4bb-pzxgj/pod?namespace=auction-ui-production&container=bot-twitter

**Comment by LukSteib** [comment]
@florianstoecker can you please take over functional testing on https://production.auction-ui.k8s.sidestream.tech

Review flow should include:
* Connect with wallet
* Restart expired auction
* auth steps
* de-auth steps
* execute an auction

**Comment by florianstoecker** [comment]
> * de-auth steps

How to do that?

**Comment by LukSteib** [comment]
> How to do that?
Execute the following commands via console:

Wallet 
`await $nuxt.$store.dispatch('authorizations/deauthorizeWallet')`

Collateral type (Example for AAVE-A)

`await $nuxt.$store.dispatch('authorizations/deauthorizeCollateral', 'AAVE-A')`

**Comment by florianstoecker** [comment]
Done on goerli + kovan:
- Connect with wallet
- auth steps
- de-auth steps
- execute an auction
Works beautifully

The restarting was only tested on kovan, since the goerli auctions will take an hour to expire.

One bug I've discovered:
For the restarting action(kovan): I do not think that this is related to this change, but seems like there is a bug with the auction time where auctions already appear as Requiring a restart even before the auction time drops to 0. 
image

**Comment by valiafetisov** [comment]
Thanks @florianstoecker, the problem is certainly unrelated. Can you please approve the PR if everything else worked?

@BracketJohn, what do you make of https://github.com/sidestream-tech/unified-auctions-ui/pull/57#issuecomment-1020987282?

**Review by florianstoecker — APPROVED** [review]

File: .drone.yml
Comment (valiafetisov) [inline .drone.yml:91]: Do we keep old registry because it allows us to keep images private?

File: .drone.yml
Comment (valiafetisov) [inline .drone.yml:202]: Is it actually required to have no spaces in the step name? (: 
```suggestion
    - name: Deploy production.auction-ui.k8s.sidestream.tech
```

**Review by valiafetisov — COMMENTED** [review]
Please revert code to the correct trigger (when the bot-twitter error is resolved) so the code can actually be approved 🙂

**Comment by BracketJohn** [comment]
> @BracketJohn, what do you make of #57 (comment)?

hm - unsure as I don't know the service. Without debugging: can this be the result of not setting the twitter env vars at all?

File: .drone.yml
Comment (BracketJohn) [reply .drone.yml:91]: No, just a value that was not yet updated. Pointed it to use new repo name!

File: .drone.yml
Comment (BracketJohn) [reply .drone.yml:202]: It's not, I would prefer it though as these names are also used in other parts of the code (like declaring dependencies on these steps). So to keep it all consistent and avoid unexpected surprises, i'd like to stay with `all-lower-case-dash-naming`

File: kubernetes/helm/values.override_production.auction-ui.k8s.sidestream.tech.yaml
Comment (valiafetisov) [inline kubernetes/helm/values.override_production.auction-ui.k8s.sidestream.tech.yaml:2]: But we should set several hosts over here, including:
- `auctions.makerdao.network`
- `unified-auctions.makerdao.com`

So the DNS switching can be done with no downtime

CC @florianstoecker: we should whitelist another origin for the production frontend infura key

**Review by valiafetisov — COMMENTED** [review]
Btw, do `ghcr.io/sidestream-tech/unified-auctions-ui/*` images publicly accessible?

File: kubernetes/helm/values.override_production.auction-ui.k8s.sidestream.tech.yaml
Comment (BracketJohn) [inline kubernetes/helm/values.override_production.auction-ui.k8s.sidestream.tech.yaml:4]: @valiafetisov note how now two hosts are defined here.

I did a test deployment via the CD pipeline in the previous commits. To test it out:
```sh
> curl 35.156.101.56 -H 'Host: auctions.makerdao.network' -v
> curl 35.156.101.56 -H 'Host: production.auction-ui.k8s.sidestream.tech' -v 
```

Both should end up redirecting you within the cluster (due to tls). If you add `-L` you will also end up visiting the page.

File: kubernetes/helm/values.override_production.auction-ui.k8s.sidestream.tech.yaml
Comment (valiafetisov) [inline kubernetes/helm/values.override_production.auction-ui.k8s.sidestream.tech.yaml:13]: So we would still need to redeploy it, since tls manager will not run itself when dns entry has changed (and therefore requiring manual edit of this file)?

File: kubernetes/helm/values.override_production.auction-ui.k8s.sidestream.tech.yaml
Comment (valiafetisov) [reply kubernetes/helm/values.override_production.auction-ui.k8s.sidestream.tech.yaml:4]: But that is only about traffic on port 80 🙁

File: kubernetes/helm/values.override_production.auction-ui.k8s.sidestream.tech.yaml
Comment (valiafetisov) [reply kubernetes/helm/values.override_production.auction-ui.k8s.sidestream.tech.yaml:13]: Same with the `unified-auctions.makerdao.com` domain, I guess

File: kubernetes/helm/values.override_production.auction-ui.k8s.sidestream.tech.yaml
Comment (florianstoecker) [reply kubernetes/helm/values.override_production.auction-ui.k8s.sidestream.tech.yaml]: which one? Currently Whitelisted are:
- https://unified-auctions.makerdao.com
- https://production.auction-ui.k8s.sidestream.tech
- https://auctions.makerdao.network/

Comment (valiafetisov) [reply kubernetes/helm/values.override_production.auction-ui.k8s.sidestream.tech.yaml]: This list is enough, thank you (:

File: kubernetes/helm/values.override_production.auction-ui.k8s.sidestream.tech.yaml
Comment (BracketJohn) [reply kubernetes/helm/values.override_production.auction-ui.k8s.sidestream.tech.yaml:13]: > So we would still need to redeploy it, since tls manager will not run itself when dns entry has changed (and therefore requiring manual edit of this file)?

No, unless I misunderstood, but: What we're doing now is listening to two hosts:
- production.auction-ui.k8s.sidestream.tech, with the TLS cert stored in `auction-ui-tls-certificate`
- auctions.makerdao.network, without any TLS cert

They have to take care of TLS termination (this is also what's going on right now with the current deployment).

File: kubernetes/helm/values.override_production.auction-ui.k8s.sidestream.tech.yaml
Comment (BracketJohn) [reply kubernetes/helm/values.override_production.auction-ui.k8s.sidestream.tech.yaml:4]: In a short term I could not come up with a better solution to proof to you that we're listening to both hosts, as we do not have control over their domain and TLS certs (see below) 😢

File: kubernetes/helm/values.override_production.auction-ui.k8s.sidestream.tech.yaml
Comment (BracketJohn) [reply kubernetes/helm/values.override_production.auction-ui.k8s.sidestream.tech.yaml:13]: > Same with the unified-auctions.makerdao.com domain, I guess

This is the first time I head about this domain, but yeah, this one is not currently listened for.

Comment (valiafetisov) [reply kubernetes/helm/values.override_production.auction-ui.k8s.sidestream.tech.yaml:13]: > They have to take care of TLS termination

I was not aware of that!

> This is the first time I head about this domain

I've mentioned it here: https://github.com/sidestream-tech/unified-auctions-ui/pull/57#discussion_r793443121
Do you want to add it in the same PR before merging it?

Comment (BracketJohn) [reply kubernetes/helm/values.override_production.auction-ui.k8s.sidestream.tech.yaml:13]: So, I've added the second domain - when we change the main environment to `unified-auctions.makerdao.com` we will have to update the deployment anyways though, to update the plausible link. But I guess that this is not super relevant, as we can just count both `unified-auctions.makerdao.com` and `auctions.makerdao.network` as production for analytics.

Comment (valiafetisov) [reply kubernetes/helm/values.override_production.auction-ui.k8s.sidestream.tech.yaml:13]: > plausible link

It's not a link, it's something like a project id! Maybe we should change the variable name to make it less confusing 😕

**Review by valiafetisov — APPROVED** [review]

**Pull request merged**
